packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -0,0 +1,153 @@
+        }
+    };
+
+    const onMouseLeave = () => {
+        setHoveredBarrier(null);
+    };
+
+    React.useEffect(() => setSelectedBarrier(barrier_1), [barrier_1]);
+
+    return (
+        <React.Fragment>
+            {is_barriers_table_expanded && (
+                <CSSTransition
+                    appear
+                    in={is_barriers_table_expanded}
+                    timeout={250}
+                    classNames={{
+                        appear: 'trade-container__barriers-table--enter',
+                        enter: 'trade-container__barriers-table--enter',
+                        enterDone: 'trade-container__barriers-table--enter-done',
+                        exit: 'trade-container__barriers-table--exit',
+                    }}
+                    unmountOnExit
+                >
+                    <Fieldset className='trade-container__fieldset trade-container__barriers-table'>
+                        <div className='trade-container__barriers-table__header'>
+                            <Text color='prominent' weight='bold' size='xs'>
+                                {localize('Barriers')}
+                            </Text>
+                            <div className='trade-container__barriers-table__icon-close' onClick={toggleBarriersTable}>
+                                <Icon icon='IcCross' />
+                            </div>
+                        </div>
+                        <div className='trade-container__barriers-table__text'>Distance to spot</div>
+                        <ThemedScrollbars>
+                            <BarriersList
+                                base_classname='trade-container__barriers-table__item'
+                                active_item_classname='trade-container__barriers-table__item--selected'
+                                className='trade-container__barriers-table__list'
+                                list={turbos_barrier_choices}
+                                selected_item={selected_barrier}
+                                onClick={changeBarrier}
+                                onMouseLeave={onMouseLeave}
+                                onMouseEnter={onMouseEnter}
+                                hover_item_classname='trade-container__barriers-table__item--highlight'
+                            />
+                        </ThemedScrollbars>
+                    </Fieldset>
+                </CSSTransition>
+            )}
+            <Fieldset
+                className='trade-container__fieldset trade-container__barriers'
+                header={localize('Barrier')}
+                is_center
+                header_tooltip={<Localize i18n_default_text={header_tooltip_text} />}
+            >
+                <div onClick={toggleBarriersTable} className='trade-container__barriers__wrapper'>
+                    <div className='trade-container__barriers-spot'>{localize('Spot')}</div>
+                    <InputField
```

Comment:
@kate-deriv i actually don't think InputField is needed here at all. we should not let a user input values, right? instead it has to be a component/tag, e.g. div, which only **displays** a selected value, and barrier list should expand when a user clicks on this displayed barrier value which will only look like an input🙂

Reply:
I tried to reuse existing components, so after user clicked on input he will see the barriers table. No problem, I'll do a simple `div`